exporter/awsemfexporter/metric_translator_test.go

```diff
@@ -1323,36 +1333,179 @@ func BenchmarkTranslateOtToCWMetricWithNamespace(b *testing.B) {
+					Description: "Counting all the spans",
+					Unit:        "Count",
+					Type:        metricspb.MetricDescriptor_CUMULATIVE_INT64,
+				},
+				Timeseries: []*metricspb.TimeSeries{},
+			},
+			{
+				MetricDescriptor: &metricspb.MetricDescriptor{
+					Name:        "spanGaugeCounter",
+					Description: "Counting all the spans",
+					Unit:        "Count",
+					Type:        metricspb.MetricDescriptor_GAUGE_INT64,
+				},
+				Timeseries: []*metricspb.TimeSeries{},
+			},
+			{
+				MetricDescriptor: &metricspb.MetricDescriptor{
+					Name:        "spanGaugeDoubleCounter",
+					Description: "Counting all the spans",
+					Unit:        "Count",
+					Type:        metricspb.MetricDescriptor_GAUGE_DOUBLE,
+				},
+				Timeseries: []*metricspb.TimeSeries{},
+			},
+			{
+				MetricDescriptor: &metricspb.MetricDescriptor{
+					Name:        "spanDoubleCounter",
+					Description: "Counting all the spans",
+					Unit:        "Count",
+					Type:        metricspb.MetricDescriptor_CUMULATIVE_DOUBLE,
+				},
+				Timeseries: []*metricspb.TimeSeries{},
+			},
+			{
+				MetricDescriptor: &metricspb.MetricDescriptor{
+					Name:        "spanDoubleCounter",
+					Description: "Counting all the spans",
+					Unit:        "Count",
+					Type:        metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION,
+				},
+				Timeseries: []*metricspb.TimeSeries{},
+			},
+		},
+	}
+	rm = internaldata.OCToMetrics(md).ResourceMetrics().At(0)
+	cwm, totalDroppedMetrics = TranslateOtToCWMetric(&rm, ZeroAndSingleDimensionRollup, cwMetricsMap, groupedCWMetricMap, "")
+	assert.Equal(b, 0, totalDroppedMetrics)
+	assert.NotNil(b, cwm)
+	assert.Equal(b, 1, len(cwm))
 
 	b.ResetTimer()
 	for n := 0; n < b.N; n++ {
-		TranslateOtToCWMetric(&rm, ZeroAndSingleDimensionRollup, "")
+		TranslateOtToCWMetric(&rm, ZeroAndSingleDimensionRollup, cwMetricsMap, groupedCWMetricMap, "")
 	}
 }
 
-func BenchmarkTranslateCWMetricToEMF(b *testing.B) {
-	cwMeasurement := CwMeasurement{
-		Namespace:  "test-emf",
-		Dimensions: [][]string{{"OTelLib"}, {"OTelLib", "spanName"}},
+func TestBatchCWMetrics(t *testing.T) {
+	cwMetricsMap := make(map[string]*CWMetrics)
+	groupedCWMetricMap := make(map[string]*GroupedCWMetric)
+
+	cwMeasurement_1 := CwMeasurement{
+		Namespace:  "eks-aoc",
+		Dimensions: [][]string{{"controller_pod", "kubernetes_node"}},
 		Metrics: []map[string]string{{
-			"Name": "spanCounter",
-			"Unit": "Count",
+			"Name": "nginx_ingress_controller_nginx_process_connections",
+			"Unit": "",
 		}},
 	}
-	timestamp := int64(1596151098037)
-	fields := make(map[string]interface{})
-	fields["OTelLib"] = "cloudwatch-otel"
-	fields["spanName"] = "test"
-	fields["spanCounter"] = 0
+	timestamp_1 := int64(1603909497679)
+	fields_1 := make(map[string]interface{})
+	fields_1["nginx_ingress_controller_nginx_process_connections"] = 0
+	fields_1["controller_pod"] = "my-nginx-ingress-nginx-controller-bbf548c86-wl84b"
+	fields_1["kubernetes_node"] = "ip-192-168-17-95.us-west-2.compute.internal"
```

Comment:
nit: I would suggest using dummy values here for test cases, like `field1` or `a` vs using these "real-world" specific values.